### DISCUSSION TITLE
New programming language in code block options - ABAP

### DISCUSSION THREAD

**Discussion by adrianobh** [zim-desktop-wiki/zim-desktop-wiki#1859] [Ideas]
I would like to suggest the inclusion of the ABAP programming language in the options that exist to include code blocks in the application.

**Comment by jaap-karssenberg** [comment]
The language support of the code blocks is part of gtksourceview, which is
a library we re-use. Please file a request there.

-- Jaap

**Comment by dehesselle** [comment]
In case you want to create a language definition yourself @adrianobh , it's not that hard - well it depends on how complex ABAP is I guess. It's an `xml` file to be placed in a specific folder, in case you are interested: https://github.com/zim-desktop-wiki/zim-desktop-wiki/issues/451#issuecomment-972393437
You can use any existing file as a reference.
